### PR TITLE
[systemtest] Check real logs for pattern in logging tests

### PR DESCRIPTION
**Pull request by im-konge**
Signed-off-by: Lukas Kral <lukywill16@gmail.com>

### Type of change

- Enhancement

### Description

Currently we are checking in our logging tests (`LoggingChangeST`) that pod contains `log4j` properties file, which is changing while we changing CMs of each components. It can be problem when the `log4j` is not used in images (by mistake) - the file itself is changed, but the logging not. This PR adds checks, that log from certain pod contains `log4j` pattern.

Also fixes issue with JSON format logging test - even when we didn't get a JSON object from log we returned `true`, which lead to wrong assertions in our checks. We had also some edge cases, where regex wasn't matching any records. I'm adding parser to `StUtils`, which processes block of text char by char, creating `JsonObject`s and then `JsonArray`.

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

File: systemtest/src/main/java/io/strimzi/systemtest/utils/StUtils.java [hunk 1]
```diff
@@ -434,4 +411,49 @@ public static void copyImagePullSecret(String namespace) {
                 .withData(Collections.singletonMap(".dockerconfigjson", pullSecret.getData().get(".dockerconfigjson")))
                 .build());
     }
+
+    private static JsonArray getJsonArrayFromLog(String log) {
```
Comment (see-quick) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/StUtils.java:415]: JFYI: If you are making the parser the great way how to make it quickly only by defining excepted structure and 
 then just generate it by `ANTLR`...

Comment (see-quick) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/StUtils.java:415]: Would be great to put here some JavaDoc what is the excepted structure of the input `log` :)

File: systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java [hunk 2]
```diff
@@ -140,6 +143,9 @@ void testJSONFormatLogging(ExtensionContext extensionContext) {
         String configMapKafkaName = "json-layout-kafka";
         String configMapCOName = Constants.STRIMZI_DEPLOYMENT_NAME;
 
+        String originalCoLoggers = kubeClient().getClient().configMaps()
+            .inNamespace(INFRA_NAMESPACE).withName(configMapCOName).get().getData().get("log4j2.properties");
```
Comment (see-quick) [inline systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java:147]:
```suggestion
.inNamespace(clusterOperator.getDeploymentNamespace()).withName(configMapCOName).get().getData().get("log4j2.properties");
```
This needs to change because when you try OLM here (it won't work)

File: systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java [hunk 3]
```diff
@@ -240,12 +246,16 @@ void testJSONFormatLogging(ExtensionContext extensionContext) {
         StUtils.checkLogForJSONFormat(namespaceName, zkPods, "zookeeper");
         StUtils.checkLogForJSONFormat(namespaceName, eoPods, "topic-operator");
         StUtils.checkLogForJSONFormat(namespaceName, eoPods, "user-operator");
+
+        // set loggers of CO back to original
+        configMapCO.getData().put("log4j2.properties", originalCoLoggers);
+        kubeClient().getClient().configMaps().inNamespace(INFRA_NAMESPACE).createOrReplace(configMapCO);
```
Comment (see-quick) [inline systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java:252]: same here

**Review by see-quick — COMMENTED** [review]
Looks good ;) 👌 just few comments  🦥

File: systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java (hunk 2 above)
Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java:147]: Completely forgot that! Good catch 🪲

File: systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java (hunk 3 above)
Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java:252]: 🪲🤲

File: systemtest/src/main/java/io/strimzi/systemtest/utils/StUtils.java (hunk 1 above)
Comment (im-konge) [reply systemtest/src/main/java/io/strimzi/systemtest/utils/StUtils.java:415]: Good to know 🤔 thanks!

File: systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java (hunk 2 above)
Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java:147]: I also changed it everywhere else in test

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=LoggingChangeST

**Review by see-quick — APPROVED** [review]

**Review by Frawless — APPROVED** [review]

**Comment by im-konge** [comment]
Failures in AZP are not related to this PR
